models/contacts.js

```diff
@@ -1,19 +1,104 @@
-// const fs = require('fs/promises')
+const { readJsonFile, writeJsonFile } = require("../utils/fileOperations");
 
-const listContacts = async () => {}
 
-const getContactById = async (contactId) => {}
+const { safeJoin } = require("../utils/path");
 
-const removeContact = async (contactId) => {}
+const contactsPath = safeJoin(__dirname, "contacts.json");
 
-const addContact = async (body) => {}
+const listContacts = async () => {
+  try {
+    const list = await readJsonFile(contactsPath);
+    return list;
+  } catch (error) {
+    console.log(error);
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const getContactById = async (contactId) => {
+  try {
+    const list = await listContacts();
+    const contact = list.find((c) => Number(c.id) === Number(contactId));
+
+    return contact;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const removeContact = async (contactId) => {
+  try {
+    const list = await listContacts();
+
+    const removedContact = getContactById(contactId);
+
+    const listWithoutRemovedContact = list.filter(
+      (c) => Number(c.id) !== Number(contactId)
+    );
+
+    await writeJsonFile(contactsPath, listWithoutRemovedContact);
+
+    return removedContact;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const addContact = async (body) => {
+  const { email, name, phone } = body;
+  try {
+    const list = await listContacts();
+
+    const lastId = list.reduce((a, b) => {
+      const prevId = Number(a.id);
+      const nextId = Number(b.id);
+      return prevId > nextId ? prevId : nextId;
+    }, 1);
```

Comment:
Tutaj tez problem z typem danych. Jezeli ciagle musisz robic `Number()`, to duzo lepiej bedzie przeksztalcic `id` w typ numeryczny na poziomie zapamietywania id (ale Ty celowo robisz z id String na L58).
Wtedy Twoje reduce mogloby wygladac ladnie tak:
`contacts.reduce((acc, c) => Math.max(acc.id, c.id), 1)`